nilearn/_utils/niimg_conversions.py

```diff
@@ -102,7 +102,23 @@ def _iter_check_niimg(niimgs, ensure_ndim=None, atleast_4d=False,
         Data type toward which the data should be converted. If "auto", the
         data will be converted to int32 if dtype is discrete and float32 if it
         is continuous.
+
+    See also
+    --------
+        check_niimg, check_niimg_3d, check_niimg_4d
     """
+    # Verify the input list is not empty, otherwise raise an error.
+    if isinstance(niimgs, list) and len(niimgs) == 0:
```

Comment:
Calling `len` on `niimgs` won't work if `niimgs` is a generator. What you can do is checking the number of loops done in the `for loop` below.